rest-client/src/main/java/org/eclipse/dataspaceconnector/registration/client/ApiClientFactory.java

```diff
@@ -40,6 +42,11 @@ private ApiClientFactory() {
     @NotNull
     public static ApiClient createApiClient(String baseUri, Function<TokenParameters, Result<TokenRepresentation>> credentialsProvider) {
         var apiClient = new ApiClient();
+        apiClient.setHttpClientBuilder(
+                apiClient.createDefaultHttpClientBuilder()
+                        .connectTimeout(Duration.ofSeconds(30))
```

Comment:
nit 1: magic numbers should be set as constant
nit 2: timeout values should be configurable from outside (e.g. config file or env variable)

Reply:
Aggred on the nit's adjusting a default value or being able to change via config is desireable. I will approve and leave this to you @cpeeyush